modules/datahandling/userrequests.go

```diff
@@ -78,10 +117,53 @@ func (f *userLookupRequest) setAbstractRequest(req *abstractRequest) {
 	f.abstractRequest = *req
 }
 
-func (f userLookupRequest) process() (response *serverMessageWrapper, notification *serverMessageWrapper, err error) {
-	// TODO
-	fmt.Printf("Recieved user lookup request from %s\n", f.SenderID)
-	return nil, nil, nil
+func (f userLookupRequest) process(db dbfs.DBFS) ([]dhClosure, error) {
+	users := make([]dbfs.UserMeta, len(f.Usernames))
+	index := 0
+	var erro error
+	for _, username := range f.Usernames {
+		usr, err := db.MySQLUserLookup(username)
+		if err != nil {
+			erro = err
+		} else {
+			users[index] = usr
+			index++
+		}
+	}
+	// shrink as needed
+	users = users[:index]
+
+	res := new(serverMessageWrapper)
+	res.Timestamp = time.Now().UnixNano()
+	res.Type = "Responce"
+
+	if len(users) < 0 {
+		res.ServerMessage = response{Status: fail, Tag: f.Tag}
+	} else {
+		if erro != nil {
+			// at least 1 value failed
+			// return what we can but
+			// tell the client whatever they don't get back failed
+			res.ServerMessage = response{
```

Comment:
hmm, that would be a bit tricky. plus unless we plan on changing some models and sending them to the client it doesn't necessarily matter at the moment